src/itwinai/torch/inference.py

```diff
@@ -117,18 +195,27 @@ def execute(
             Dict[str, Any]: maps each item ID to the corresponding predicted
                 value(s).
         """
+        self._init_distributed_strategy()
         if model is not None:
             # Overrides existing "internal" model
             self.model = model
 
-        test_dataloader = self.test_dataloader_class(
-            test_dataset, **self.test_dataloader_kwargs
+        self.create_dataloaders(
+            inference_dataset=inference_dataset
         )
 
+        self.distribute_model()
+
+        if self.logger:
+            self.logger.create_logger_context(rank=self.strategy.global_rank())
+            hparams = self.config.model_dump()
+            hparams['distributed_strategy'] = self.strategy.__class__.__name__
+            self.logger.save_hyperparameters(hparams)
+
         all_predictions = dict()
-        for samples_ids, samples in test_dataloader:
+        for ids, (samples_ids, samples) in enumerate(self.inference_dataloader):
             with torch.no_grad():
-                pred = self.model(samples)
+                pred = self.model(samples.to(self.device))
```

Comment:
I can only leave a comment on this line as GH is not letting me to review lines that were not changed.
I would suggest to move the block from line 215 to line 226 to a separate method called `predict`, which is the equivalent of the `train` method in the trainer